### PR TITLE
Upgrade worker-dom to 0.12.3

### PR DESCRIPTION
**Pull request by choumx**
Tested on max/min builds.

### CODE REVIEW

File: build-system/tasks/update-packages.js [hunk 1]
```diff
@@ -119,15 +119,13 @@ function patchRegisterElement() {
 }
 
 /**
- * Closure Compiler doesn't recognize .mjs extension yet, so copy the file to
- * have a .js file extension.
+ * transformEs6Packages() doesn't work on subdirectories of dist/,
+ * and I can't figure out which option enables it.
```
Comment (choumx) [inline build-system/tasks/update-packages.js]: @rsimha I noticed you added `transformEs6Packages()` as part of Babel 7 migration. Is `global: true` problematic now? We still use it in a few places.

File: build-system/tasks/extension-helpers.js [hunk 2]
```diff
@@ -457,11 +457,11 @@ function buildExtensionJs(path, name, version, latestVersion, options) {
   ).then(() => {
     // Copy @ampproject/worker-dom/dist/worker.safe.js to the dist/ folder.
     if (name === 'amp-script') {
-      // TODO(choumx): Compile this when worker-dom externs are available.
-      const dir = 'node_modules/@ampproject/worker-dom/dist/';
+      const dir = 'node_modules/@ampproject/worker-dom/dist/amp/worker/';
```
Comment (jridgewell) [inline build-system/tasks/extension-helpers.js]: Weird directory structure.

File: build-system/tasks/update-packages.js (hunk 1 above)
Comment (rsimha) [reply build-system/tasks/update-packages.js]: I documented this in item 7 of https://github.com/ampproject/amphtml/pull/18574#issue-220819628. IIRC, `global: true` causes _all_ packages in `node_modules` to be transformed, which at the time sounded more risky than transforming packages as needed. See [this](https://github.com/babel/babelify#why-arent-files-in-node_modules-being-transformed) for more details.

Comment (choumx) [reply build-system/tasks/update-packages.js]: OK, wasn't sure if there was a concrete problem. Going to see if we can avoid the hack in this PR by enabling `global` but ignoring `devDependencies` paths.

File: build-system/tasks/extension-helpers.js (hunk 2 above)
Comment (choumx) [reply build-system/tasks/extension-helpers.js]: Fixed in #22927.

**Pull request merged**